class/class.2.1/hello.cpp

```diff
@@ -0,0 +1,84 @@
+#include <iostream>
+#include <fstream>
+#include <unordered_map>
+#include <string>
+
+const std::string FILENAME = "user.txt";
+
+class UserStats {
+public:
+    void load() {
+        std::ifstream infile(FILENAME);
+        if (!infile) {
+            std::cerr << "Failed to open file for reading.\n";
+            return;
+        }
+
+        std::string name;
+        int count;
+        while (infile >> name >> count) {
+            stats[name] = count;
+        }
+        std::cout << "Loaded stats from file.\n";
+    }
+
+    void save() const {
+        std::ofstream outfile(FILENAME);
+        if (!outfile) {
+            std::cerr << "Failed to open file for writing.\n";
+            return;
+        }
+
+        for (const auto& [name, count] : stats) {
+            outfile << name << " " << count << "\n";
+        }
+        std::cout << "Saved stats to file.\n";
+    }
+
+    void reset(const std::string& name) {
+        if (name == "bread") {
+            stats.clear();
+            std::cout << "All history cleared.\n";
+        } else {
+            stats.erase(name);
+            std::cout << "Statistics reset for " << name << ".\n";
+        }
+    }
+
+    void greet(const std::string& name) {
+        if (stats.find(name) == stats.end()) {
+            stats[name] = 1;
```

Comment:
if key is not present then stats[name] will be initialised with 0 therefore in both cases we can just do stats[name]++ and if we will combine it with a comment above then it will be something like
```
int incrementFor(const std::string& name) {
    stats[name]++;
    return stats[name];
}
```